### DISCUSSION TITLE
Newbie-friendly control of python environments

### DISCUSSION THREAD

**Discussion by JoshMerfeld** [posit-dev/positron#4148] [Feedback]
Coming from R, one of the biggest issues I had with getting started in python was just downloading it (ugh!) and getting the hang of environments. I think something that could really help Positron with getting newbies into Python is a point-and-click interface to
- Create a new environment using e.g. conda
- Install packages into that environment
- Activate/deactivate that environment (this already happens somewhat with the automatic detection of existing environments when using "open folder")

I have always said that I thought more people would switch to python if it had an IDE (and installation/package manager!) as user-friendly as R. Positron strikes me as the closest to that, so far.

**Comment by jthomasmock** [comment]
Thanks Josh for the commentary here!

You may want to upvote: https://github.com/posit-dev/positron/issues/1391 and the idea of a "Enhanced Environments Management pane"

We _do_ have some initial support for creating a basic Python environment with `.venv` or `conda` via the New Project flow:
![image](https://github.com/user-attachments/assets/129c89e6-f2b4-41ca-8cfd-4b8088b6e4b5)

**Reply by JoshMerfeld** [reply]
Thanks, Tom! Yes, this is definitely part of the way there. I upvoted the other discussion because that's definitely along the lines of what I was thinking. I lvoe the idea of a "env pane" that has information on packages, versions, etc.! (Plus a point-and-click interface for the newbies out there.)

**Reply by ntluong95** [reply]
Is this possible to have set up a new Python environment with Poetry

**Reply by ntluong95** [reply]
It should be something like this: https://www.jetbrains.com/help/pycharm/poetry.html

**Reply by juliasilge** [reply]
Not yet, no! I'll add this feature request to #2725.

**Reply by martinhulin** [reply]
Will this enable to create conda envs for R projects?

**Reply by juliasilge** [reply]
Ah no, not currently @martinhulin 

We support using renv for R projects but not other dependency management systems:

![Screenshot 2024-07-30 at 8 04 09 AM](https://github.com/user-attachments/assets/6731fad0-c97b-4675-a8ee-8f7d3b6ca7c6)

We can collect feedback about what other systems folks prefer to use in the New Project Wizard, such as conda.

**Reply by martinhulin** [reply]
Thanks Julia! Just to clarify, does it mean if i have r within a conda or pixi env, positron will not be able to find it (as a kernel or interpreter?) and if so is there any particular reason for this (can one manully setup the path to the interpreter within conda/pixi)? Will this be true also when eg. pixi will be enabled, i.e. ability of Positron to find the interpreter (python, R, julia) within an environment? Problem with renv is it does not solve what conda or pixi can, ie. the entire software dependecy not just package dependency, as this is so critical in industry. I am not even able to install packages outside conda or pixi, so R users face huge pushback and difficulty. So would welcome any progress to conda/pixi + R envs.

**Reply by juliasilge** [reply]
There are a handful of issues on this so far, which you might want to add your experience and feedback to:

- https://github.com/posit-dev/positron/issues/2235
- https://github.com/posit-dev/positron/issues/3724
- https://github.com/posit-dev/positron/issues/4047
